cmd/frontend/db/schemadoc/main.go

```diff
@@ -19,21 +21,72 @@ import (
 // PGPORT, PGUSER etc. env variables must be set to run this script.
 //
 // First CLI argument is an optional filename to write the output to.
-func main() {
+func generate(log *log.Logger) (string, error) {
 	const dbname = "schemadoc-gen-temp"
-	_ = exec.Command("dropdb", dbname).Run()
-	if out, err := exec.Command("createdb", dbname).CombinedOutput(); err != nil {
-		log.Fatalf("createdb: %s, %v", out, err)
+
+	var (
+		dataSource string
+		run        func(cmd ...string) (string, error)
+	)
+	// If we are using pg9.6 use it locally since it is faster (CI \o/)
+	versionRe := regexp.MustCompile(fmt.Sprintf(`\b%s\b`, regexp.QuoteMeta("9.6")))
+	if out, _ := exec.Command("psql", "--version").CombinedOutput(); versionRe.Match(out) {
+		dataSource = "dbname=" + dbname
+		run = func(cmd ...string) (string, error) {
+			c := exec.Command(cmd[0], cmd[1:]...)
+			c.Stderr = log.Writer()
+			out, err := c.Output()
+			return string(out), err
+		}
+		_ = exec.Command("dropdb", dbname).Run()
+		defer exec.Command("dropdb", dbname).Run()
+	} else {
+		log.Printf("Running PostgreSQL 9.6 in docker since local version is %s", strings.TrimSpace(string(out)))
+		_ = exec.Command("docker", "rm", "--force", dbname).Run()
+		server := exec.Command("docker", "run", "--rm", "--name", dbname, "-p", "5433:5432", "postgres:9.6")
+		if err := server.Start(); err != nil {
+			return "", err
+		}
+
+		defer func() {
+			_ = server.Process.Kill()
+			_ = exec.Command("docker", "kill", dbname).Run()
+			_ = server.Wait()
+		}()
+
+		time.Sleep(1 * time.Second)
+		dataSource = "postgres://postgres@localhost:5433/postgres?dbname=" + dbname
+		run = func(cmd ...string) (string, error) {
+			cmd = append([]string{"exec", "-u", "postgres", dbname}, cmd...)
+			c := exec.Command("docker", cmd...)
+			c.Stderr = log.Writer()
+			out, err := c.Output()
+			return string(out), err
+		}
+
+		attempts := 0
+		for {
+			attempts++
+			if err := exec.Command("pg_isready", "-U", "postgres", "-d", dbname, "-h", "localhost", "-p", "5433").Run(); err == nil {
+				break
+			} else if attempts > 30 {
+				return "", fmt.Errorf("gave up waiting for pg_isready: %w", err)
```

Comment:
Not sure if this is better:

```suggestion
				return "", fmt.Errorf("gave up waiting after 30s attempt for pg_isready: %w", err)
```